File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/scenario/ScenarioListener.java
package com.gmail.borlandlp.minigamesdtools.arena.scenario;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerDeathLocalEvent;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerJoinLocalEvent;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerKilledLocalEvent;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerLeaveLocalEvent;

public class ScenarioListener implements ArenaEventListener {
    private ScenarioChainController scenarioChainController;

    public ScenarioListener(ScenarioChainController scenarioChainController) {
        this.scenarioChainController = scenarioChainController;
    }

    public void onPlayerDeath(ArenaPlayerDeathLocalEvent event) {

    }

    public void onPlayerKilled(ArenaPlayerKilledLocalEvent event) {

    }

    public void onPlayerLeave(ArenaPlayerLeaveLocalEvent event) {

    }

    public void onPlayerJoin(ArenaPlayerJoinLocalEvent event) {

    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/scenario/ScenarioListener.kt
package com.gmail.borlandlp.minigamesdtools.arena.scenario

import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerDeathLocalEvent
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerJoinLocalEvent
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerKilledLocalEvent
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerLeaveLocalEvent

class ScenarioListener(private val scenarioChainController: ScenarioChainController) : ArenaEventListener {
    fun onPlayerDeath(event: ArenaPlayerDeathLocalEvent) {}
    fun onPlayerKilled(event: ArenaPlayerKilledLocalEvent) {}
    fun onPlayerLeave(event: ArenaPlayerLeaveLocalEvent) {}
    fun onPlayerJoin(event: ArenaPlayerJoinLocalEvent) {}
}